### ISSUE TITLE
Clean up project directories by hiding intermediate files

### ISSUE THREAD

**Issue by mcgordonite** [Open-CMSIS-Pack/devtools#1301]
**Describe The Problem To Be Solved**

Toolbox creates a large number of intermediate files in the project directories, including cbuild.yml and cprj files. This makes it hard for new users to understand what they should edit and what is generated.

![Solution Directory](https://github.com/Open-CMSIS-Pack/devtools/assets/12599255/1679e986-1e42-4d26-9d65-f37f8f9bd604)

**(Optional): Suggest A Solution**

Is it possible to generate the cbuild and cprj files in the intermediate directory instead?

⚠️ This could be a breaking change if applications are relying on reading the intermediate files.

**Comment by jkrech** [comment]
Stating the obvious: `myproject.cbuild-idx.yml` needs to stay side by side to the `myproject.csolution.yml` file for tools to located the *.cbuild files.

**Comment by brondani** [comment]
To place such files outside the project file tree, including cbuild and cprj files as well as tmp and out directories, `csolution` and `cbuild` tools have the `--output` option:
```
$ cbuild --help
  -O, --output arg         Set directory for all output files
```

**Comment by mcgordonite** [comment]
> To place such files outside the project file tree, including cbuild and cprj files as well as tmp and out directories, csolution and cbuild tools have the --output option

This is a good point, but it is not the "default" behaviour encouraged by the tools. New users will see the new files until they discover the `--output` options.

**Comment by ReinhardKeil** [comment]
I don't think it is required to place `out` outside of the project file tree.  It is just the default location of the cbuild.yml (and CPRJ) files that we should look at.

**Comment by jkrech** [comment]
@mcgordonite Making the `out` directory the default destination directory for `*.cbuild.yml` and `*.cprj`  is captured in #1305. Can we close this issue to avoid duplicating topics.

**Comment by jkrech** [comment]
To be solved by #1322
